src/qemu.rs

```diff
@@ -655,8 +655,28 @@ impl Qemu {
         let qmp_sock = gen_sock("qmp");
         let command_sock = gen_sock("cmdout");
         let (init, guest_init) = gen_init(&target.rootfs).context("Failed to generate init")?;
+        let program = target
+            .qemu_command
+            .unwrap_or_else(|| format!("qemu-system-{}", target.arch));
+
+        // Confirm QEMU exists first
+        if let Err(e) = Command::new(&program)
+            .arg("--help")
+            .stdout(Stdio::null())
+            .stderr(Stdio::null())
+            .spawn()
+        {
+            if let ErrorKind::NotFound = e.kind() {
+                return Err(e).context(format!(
+                    "Did not find QEMU binary {program}. Make sure QEMU is installed"
+                ));
+            } else {
+                warn!("Failed to verify that qemu is installed due to error, continuing... {e}");
+            }
+        }
```

Comment:
Mind pulling this out into a helper function? Just so the main logic in this function is a bit more compact

Reply:
Yep! Let me do that. I'll take a little bit to continue, I need to wait for a thumbs up from work about a couple things, but I'll get back to you

Reply:
Sorry for the delay. Lmk if this is waht you meant